changelog/pending/20230304--cli--add-json-flag-to-pulumi-whoami-to-emit-output-as-json.yaml

```diff
@@ -0,0 +1,4 @@
+changes:
```

Comment:
I kept each SDK commit separately - if you'd prefer I squash them, let me know!

Reply:
Thanks for the separate commits! Appreciate it.